### ISSUE TITLE
Use of moved value on primitives

### ISSUE THREAD

**Issue by reesmanp** [intellij-rust/intellij-rust#3568]
## Environment

* **Intellij-Rust plugin version:** v0.2.94.2119-183
* **Rust toolchain version:** 1.26.0
* **IDE name and version:** CLion 2018.3.4
* **Operating system:** Windows 10

## Problem description
Error in IDE says that there is a `Use of moved value` on a primitive (i64). Or at least I think that's the issue. It might be referencing the struct itself.

## Steps to reproduce

A simple example is:

```rust
struct Test {
    x: i64
}

impl Test {
    fn test_func(&mut self) {
        self.x = self.x + 1;
    }
}
```

The underline is under `self`. Adding the following function to the `impl` above also gets the same error:

```rust
fn another_test(&self) -> Self {
    Self {
        x: self.x
    }
}
```

Again the underline is under `self`.

Screenshots of the error from the IDE itself:

![f32](https://user-images.githubusercontent.com/5932381/54733026-6f690380-4b54-11e9-937a-4e5c6fb5e981.PNG)
![invert](https://user-images.githubusercontent.com/5932381/54733029-7263f400-4b54-11e9-99aa-89fef1173fc5.PNG)

**Comment by reesmanp** [comment]
Welp, I just upgraded to the newest rust version (I apparently didn't even have rustup on my machine) and everything magically works as expected.

**Comment by David-OConnor** [comment]
I'm having this trouble. On newest Jetbrains Plugin, newest rust, and cleared the cache.

**Comment by Undin** [comment]
@David-OConnor Could you create new issue and provide a code example where you faced this error with concrete versions of your environment (rustc, intellij rust plugin, etc.)?

**Comment by Allen-Webb** [comment]
I ran into this with the code of the following structure:

let limit = 1024u64;
function_that_takes_u64_twice(limit, limit)?;

I am using 0.2.123.3132-193

**Comment by Undin** [comment]
@Allen-Webb `File | Invalidate Caches/Restart` action may help.
Also, you use a quite old plugin version. To get the latest build, update your IDE to at least 2020.3

**Comment by Allen-Webb** [comment]
2020.3 isn't available to me, but updating to 2020.1 resolved that issue as well as another one I was having.
